src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java

```diff
@@ -41,12 +47,37 @@ public List<CaseEvent> handledEvents() {
     }
 
     private CallbackResponse submitSuccessfulMediation(CallbackParams callbackParams) {
+        List<Element<ManageDocument>> updatedManageDocumentsList =
+            updateManageDocumentsListWithMediationAgreementDocument(callbackParams);
+
         CaseData caseDataUpdated = callbackParams.getCaseData().toBuilder()
             .businessProcess(BusinessProcess.ready(MEDIATION_SUCCESSFUL))
+            .manageDocuments(updatedManageDocumentsList)
             .build();
+
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseDataUpdated.toMap(objectMapper))
             .state(CASE_STAYED.name())
             .build();
     }
+
+    private List<Element<ManageDocument>> updateManageDocumentsListWithMediationAgreementDocument(
+        CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<Element<ManageDocument>> manageDocumentsList = caseData.getManageDocumentsList();
+        Optional<MediationAgreementDocument> mediationAgreementDocument = Optional.ofNullable(caseData)
```

Comment:
how come caseData could be null? perhaps it should be Optional.ofNullable(casdata.getMediation()).map(Mediation::getMediationSuccessful).map(MediationSuccssful::getMediationAgreement)